controllers/storagecluster/provider_server.go

```diff
@@ -279,6 +316,61 @@ func GetProviderAPIServerSecret(instance *ocsv1.StorageCluster) *corev1.Secret {
 	}
 }
 
+func GetProviderAPIServerPermissionResources(instance *ocsv1.StorageCluster) (*rbacv1.Role, *rbacv1.RoleBinding, *corev1.ServiceAccount) {
+
+	role := &rbacv1.Role{
```

Comment:
why do we need to create these resources in golang code and not use .yaml files like other resources?

Reply:
done